### ISSUE TITLE
Interfaces not enforcing parameters correctly.

### ISSUE THREAD

**Issue by wdamien** [Microsoft/TypeScript#9794]
**TypeScript Version:**  1.8.10

**Code**

```ts
interface IFoo {
	bar(a:string);
}

class Foo implements IFoo {
	public bar() { } // Should cause an error,  since we're not accepting a parameter here.
}

```

**Expected behavior:**
- The compiler should display an error for `class Foo` since it doesn't satisfy the interface IFoo.

**Actual behavior:** 
- No error on the class declaration.

**Comment by sandersn** [comment]
See [this FAQ answer](https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters) &mdash; a function with fewer parameters is assignable to one with more. The example there is for standalone functions, but Typescript checks whether `Foo` satisfies `IFoo` the same way: by trying to assign `Foo.bar` to `IFoo.bar`.
